pennylane/ops/op_math/sum.py

```diff
@@ -298,6 +285,36 @@ def simplify(self, cutoff=1.0e-12) -> "Sum":  # pylint: disable=arguments-differ
             return Sum(*new_summands) if len(new_summands) > 1 else new_summands[0]
         return qml.s_prod(0, qml.Identity(self.wires))
 
+    def terms(self):
+        r"""Representation of the operator as a linear combination of other operators.
+
+        .. math:: O = \sum_i c_i O_i
+
+        A ``TermsUndefinedError`` is raised if no representation by terms is defined.
+
+        Returns:
+            tuple[list[tensor_like or float], list[.Operation]]: list of coefficients :math:`c_i`
+            and list of operations :math:`O_i`
+        """
+        # try using pauli_rep:
+        if pr := self.pauli_rep:
+            pr.simplify()
+            ops = [pauli.operation() for pauli in pr.keys()]
+            return list(pr.values()), ops
```

Comment:
So actually in the vast majority of cases, this method is pretty straightforward 🎉

Reply:
Yup, note however that this is doing all simplifications! Doing it fast with pauli reps without those simplifications would actually be really tricky/impossible in PauliSentence's current form

Reply:
How come this is the case? (regarding simplifications)

Reply:
The simplifications I am talking about are for example automatically summing together equal terms.
This is automatically done in `PauliSentence`
```
>>> X0 = PauliWord({0:"X"})
>>> PauliSentence({X0 : 1}) + PauliSentence({X0 : 1})
2 * X(0)
```

As far as I understand that is even a necessity since you couldnt have a dictionary with duplicate keys (its just overwriting the former with the latter):
```
>>> PauliSentence({X0 : 1, X0 : 1})
1 * X(0)
```

Reply:
> The reason I ask is that the old Hamiltonian would return terms without any simplification iirc, so am wondering if we could do the same thing here (and wondering why it's not possible/if this would help mitigate the slowdown)

I think those are two separate things that kind of interplay here.
Doing things with pauli reps is usually much faster, and they have those simplifications baked in.
Doing things with opmath could leave those simplifications for faster performance but would overall be slower, still.

For the numbers I posted: `qml.Hamiltonian` is as quick to return coeffs and ops because this is the literal definition of the operator, the way `Sum` is constructed we always have to do _some_ processing unfortunately.

Reply:
Got it, thanks!

Reply:
So to summarize: We can either have fast Pauli arithmetic, or fast ability to extract hamiltonian terms, but not both. Very similar to what you see with Sparse matrix representations (COO, CSC, CSR, etc.). Each representation has advantages and disadvantages, and if you need to convert between them you pay a computational price.

Long term: Makes me wonder if we should be upfront with this with users, retain a class for easily extracting terms/coeffs/etc that can be converted to/from PauliSentences, depending on what use case a user needs.

Reply:
> We can either have fast Pauli arithmetic, or fast ability to extract hamiltonian terms, but not both.

I would actually argue that for operators consisting purely of Paulis, you can also reasonably quickly extract `coeffs` and `ops`. There is a small overhead from the conversion. Its much more expensive if any operator is non-pauli because you cannot go down the pauli_rep route anymore.

> erms/coeffs/etc that can be converted to/from PauliSentences

Like `qml.Hamiltonian`? 🙈 😅